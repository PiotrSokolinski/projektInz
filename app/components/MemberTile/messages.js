/*
 * MemberTile Messages
 *
 * This contains all the text for the MemberTile component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.MemberTile'

export default defineMessages({
  tel: {
    id: `${scope}.tel`,
    defaultMessage: 'Tel. ',
  },
})
