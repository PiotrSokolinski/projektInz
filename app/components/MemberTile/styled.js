import styled from 'styled-components'
import { Colors } from 'themes'
import { UserDelete } from 'styled-icons/typicons/UserDelete'

export const Delete = styled(UserDelete)`
  margin-right: 15px;
  cursor: pointer;
  color: ${Colors.ecstasy};
`

export const Container = styled.div`
  border: 1px solid ${Colors.edward};
  border-radius: 15px;
  background: ${Colors.white};
  margin: 15px 10px;
  height: 75px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 10px;
`
export const InfoWrapper = styled.div`
  display: flex;
  align-items: center;
`

export const FullName = styled.div`
  margin-left: 20px;
  font-size: 19px;
  font-weight: bold;
  color: ${Colors.black};
  span {
    margin-left: 5px;
  }
`
export const Number = styled.div`
  margin-left: 30px;
  font-size: 15px;
  color: ${Colors.black};
`
