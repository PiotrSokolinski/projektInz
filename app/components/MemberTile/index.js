/**
 *
 * MemberTile
 *
 */

import React, { useState } from 'react'
import PropTypes from 'prop-types'
import remove from 'lodash/remove'
import { FormattedMessage, injectIntl } from 'react-intl'
import { compose, Mutation } from 'react-apollo'

import UserAvatar from 'components/UserAvatar'
import ConfirmationBox from 'components/ConfirmationBox'
import appLocalStorage from 'utils/localStorage'
import Spinner from 'components/Spinner'
import InformationBox from 'components/InformationBox'
import { formatGraphqlErrors } from 'utils/formatGraphqlErrors'

import messages from './messages'
import * as Styled from './styled'
import DELETE_USER_FROM_GROUP from './deleteUser.gql'
// import GET_GROUP_INFO from '../../containers/Dashboard/getGroupInfo.gql'

const MemberTile = ({
  member: { id, role, avatarUrl, firstName, lastName, number },
  intl,
  deleteUserAction,
  loading,
  errors,
}) => {
  const [isModalVisible, setIsModalVisible] = useState(false)
  const openModal = () => setIsModalVisible(true)
  const closeModal = () => setIsModalVisible(false)
  const currentUser = appLocalStorage.getSession()
  const removeUser = async () => {
    await deleteUserAction({
      variables: {
        id,
      },
    })
  }
  return (
    <Styled.Container>
      <Styled.InfoWrapper>
        <UserAvatar size="middle" image={avatarUrl} />
        <Styled.FullName>
          <span>{firstName}</span>
          <span>{lastName}</span>
        </Styled.FullName>
        {/* <Styled.Number>
          <b>
            <FormattedMessage {...messages.tel} />
          </b>
          {number}
        </Styled.Number> */}
      </Styled.InfoWrapper>
      {currentUser.id !== id && currentUser.role === 'Administrator' && <Styled.Delete size="22" onClick={openModal} />}
      <ConfirmationBox
        avatarUrl={avatarUrl}
        title={intl.formatMessage(messages.modalTitle)}
        description={`${intl.formatMessage(messages.modalDescription)} ${firstName} ${lastName}`}
        visible={isModalVisible}
        onClose={closeModal}
        error={errors}
        loading={loading}
        action={removeUser}
      />
    </Styled.Container>
  )
}

MemberTile.propTypes = {
  member: PropTypes.object,
  intl: PropTypes.object,
  deleteUserAction: PropTypes.func,
  loading: PropTypes.bool,
  errors: PropTypes.array,
}

const withDeleteMutation = Component => props => (
  <Mutation
    mutation={DELETE_USER_FROM_GROUP}
    // update={(store, { data: { removeUserFromGroup } }) => {
    //   const data = store.readQuery({ query: GET_GROUP_INFO, variables: { id: 1, current: true } })
    //   remove(data.group.members, member => member.id === removeUserFromGroup.id)
    //   store.writeQuery({
    //     query: GET_CURRENT_GROUP_TASKS,
    //     data,
    //   })
    // }}
    refetchQueries={['getGroupInfo']}
  >
    {(mutate, { loading, error }) => (
      <Component {...props} deleteUserAction={mutate} loading={loading} errors={formatGraphqlErrors(error)} />
    )}
  </Mutation>
)

export default compose(
  injectIntl,
  withDeleteMutation,
)(MemberTile)
