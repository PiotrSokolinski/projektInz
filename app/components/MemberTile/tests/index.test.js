/**
 *
 * Tests for MemberTile
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import MemberTile from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<MemberTile />', () => {
  const member = {
    avatarUrl: '',
    firstName: 'Piotr',
    lastName: 'Sokolinski',
    number: '123123123',
  }

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemberTile member={member} />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <MemberTile member={member} />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
