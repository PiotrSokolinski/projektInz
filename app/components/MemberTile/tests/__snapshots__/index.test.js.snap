// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemberTile /> Should render and match the snapshot 1`] = `
<div
  class="styled__Container-tas46t-1 jCWCTG"
>
  <div
    class="styled__InfoWrapper-tas46t-2 jOqEjB"
  >
    <div
      class="middle styled__Container-vinm4t-0 gDIJTt"
    >
      <img
        alt="avatar"
        class="avatar-image"
        src="IMAGE_MOCK"
      />
    </div>
    <div
      class="styled__FullName-tas46t-3 kDzbrN"
    >
      <span>
        Piotr
      </span>
      <span>
        Sokolinski
      </span>
    </div>
    <div
      class="styled__Number-tas46t-4 hJzDbi"
    >
      <b>
        <span>
          Tel. 
        </span>
      </b>
      123123123
    </div>
  </div>
  <svg
    aria-hidden="true"
    class="styled__Delete-tas46t-0 ftpdIp StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    height="22"
    viewBox="0 0 24 24"
    width="22"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M21 14h-6a1 1 0 110-2h6a1 1 0 110 2zm-7-5c0 1.381-.56 2.631-1.464 3.535C11.631 13.44 10.381 14 9 14s-2.631-.56-3.536-1.465C4.56 11.631 4 10.381 4 9s.56-2.631 1.464-3.535C6.369 4.56 7.619 4 9 4s2.631.56 3.536 1.465A4.984 4.984 0 0114 9zm-5 6c-3.75 0-6 2-6 4 0 1 2.25 2 6 2 3.518 0 6-1 6-2 0-2-2.354-4-6-4z"
    />
  </svg>
</div>
`;
